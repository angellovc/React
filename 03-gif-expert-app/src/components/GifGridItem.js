import React from 'react';
import '../index.css';
import PropTypes from 'prop-types';

const GifGridItem = ({id, title, url}) => {
  return (
    <div className="card-image animate__animated animate__fadeIn">
      <img src={url} alt={title}></img>
      <p>{title}</p>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}

export default GifGridItem

