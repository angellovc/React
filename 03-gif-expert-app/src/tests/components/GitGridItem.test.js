import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe('<GitGridItem /> Tests', () => {

    const title = 'This is a Title';
    const url = 'https://localhost/something.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);


    test('Snapshot must match', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Paragraph must be created using the title', () => {
        const paragraph = wrapper.find('p').text().trim();
        expect(paragraph).toBe(title);
    });

    test('img src and alt must be equal url and title', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('Div must contain animate__fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.props().className;
        expect(className?.includes('animate__fadeIn')).toBe(true);
    });

})
