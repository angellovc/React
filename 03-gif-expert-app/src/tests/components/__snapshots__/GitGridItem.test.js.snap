// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GitGridItem /> Tests Snapshot must match 1`] = `
<div
  className="card-image animate__animated animate__fadeIn"
>
  <img
    alt="This is a Title"
    src="https://localhost/something.jpg"
  />
  <p>
    This is a Title
  </p>
</div>
`;
